snippets/rust.snippets

```diff
@@ -19,32 +19,53 @@ snippet bench "Bench function" b
 			${0}
 		})
 	}
-snippet new
+snippet new "Constructor function"
 	pub fn new(${2}) -> ${1:Name} {
-		${0}return $1 { ${3} };
+		$1 { ${3} };
 	}
-snippet main
+snippet main "Main function"
 	pub fn main() {
 		${0}
 	}
-snippet let
-	let ${1:name: type} = ${2};
-snippet let
-	let mut ${1:name: type} = ${2};
-snippet pln
+snippet let "let variable declaration"
+	let ${1:name}${2:: ${3:type}} = ${4};
+snippet letm "let mut variable declaration"
+	let mut ${1:name}${2:: ${3:type}} = ${4};
+snippet pln "println!"
 	println!("${1}");
-snippet pln,
+snippet pln, "println! with format param"
 	println!("${1}", ${2});
-snippet ec
+
+# Modules
+snippet ec "extern crate"
 	extern crate ${1:sync};
-snippet ecl
-	#![feature(phase)]
-	#[phase(plugin, link)] extern crate log;
+snippet ecl "extern crate log"
+	#[macro_use]
+	extern crate log;
```

Comment:
This construct has changed because macros are no longer feature-gated in Rust. See the [official log doc](http://doc.rust-lang.org/log/log/index.html).